src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1664,7 +1665,19 @@ private function addHttpClientRetrySection()
                         ->end()
                         ->prototype('integer')->end()
                         ->info('A list of HTTP status code that triggers a retry')
-                        ->defaultValue([423, 425, 429, 500, 502, 503, 504, 507, 510])
+                        ->defaultValue(\class_exists(HttpStatusCodeDecider::class) ? HttpStatusCodeDecider::STATUS_CODES_DECISION : [])
```

Comment:
I might be reading this wrong. But this contains an array like `[500 => true]`, if we then just takes the values, we get an array with booleans.

Reply:
Good catch. Fixed (+adds test)